app/services/HCAService.js

```diff
@@ -0,0 +1,264 @@
+import Yaml from 'js-yaml';
+import PushNotification from 'react-native-push-notification';
+import RNFetchBlob from 'rn-fetch-blob';
+
+import { AUTHORITIES_LIST_URL } from '../constants/authorities';
+import {
+  AUTHORITY_SOURCE_SETTINGS,
+  HCA_AUTO_SUBSCRIPTION,
+} from '../constants/storage';
+import { GetStoreData, SetStoreData } from '../helpers/General';
+import languages from '../locales/languages';
+import { LocationData } from './LocationService';
+
+/**
+ * Singleton class to interact with health care authority data
+ */
+class HCAService {
+  /**
+   * Fetches the raw YAML file containing a list of all
+   * of the registered Health Care Authorities.
+   * Saves the response as a cached file for performance.
+   * @returns {void}
+   */
+  async fetchAuthoritiesYaml() {
+    return await RNFetchBlob.config({
+      fileCache: true,
+    }).fetch('GET', AUTHORITIES_LIST_URL);
+  }
+
+  /**
+   * Fetches the list of all registed Health Care Authorities
+   * @returns {Array} List of health care authorities from the global registry
+   */
+  async getAuthoritiesList() {
+    let authorities = [];
+
+    try {
+      const result = await this.fetchAuthoritiesYaml();
+      const list = await RNFetchBlob.fs.readFile(result.path(), 'utf8');
+      authorities = Yaml.safeLoad(list).Authorities;
+    } catch (err) {
+      console.log(err);
+    }
+
+    return authorities;
+  }
+
+  /**
+   * Get the list of Health Care Authorities that a user has saved
+   * @returns {Array} List of health care authorities from storage
+   */
+  async getUserAuthorityList() {
+    return await GetStoreData(AUTHORITY_SOURCE_SETTINGS, false);
+  }
+
+  /**
+   * Takes an array of one or more authorities and adds it to the list
+   * in storage that the user is subscribed to.
+   * @param {newAuthorities} Array healthcare authoritiy objects
+   * @returns void
+   */
+  async appendToAuthorityList(newAuthorities) {
+    const authorities = (await this.getUserAuthorityList()) || [];
+    await SetStoreData(AUTHORITY_SOURCE_SETTINGS, [
+      ...authorities,
+      ...newAuthorities,
+    ]);
+  }
+
+  /**
+   * Checks if a user has saved any Health Care Authorities
+   *
+   * @returns {boolean}
+   */
+  async hasSavedAuthorities() {
+    const authorities = await this.getUserAuthorityList();
+    return authorities && authorities.length > 0;
+  }
+
+  /**
+   * Alerts a user that there are new Healthcare Authorities in their region.
+   * Includes information on the number of Authorities in their current location.
+   *
+   * @param {numAuthorities} Number number of new authorities
+   * @returns {void}
+   */
+  async pushAlertNewAuthoritesFromLoc(numAuthorities) {
```

Comment:
async needed? if so await the localNotification

Reply:
Since these tasks are only triggered by BackgroundServices I don't think it makes sense to block on them. I'll remove the `async` - good catch.